File: leetcode/3392/c/main.c
int countSubarrays(int* nums, int numsSize) {
    int cnt = 0;
    for (int i = 0; i < numsSize - 2; i++) {
        int mid = nums[i + 1] / 2;
        if (mid * 2 == nums[i + 1] && (nums[i] + nums[i + 2]) == mid) {
            cnt++;
        }
    }

    return cnt;
}

File: leetcode/3392/rs/src/main.rs
struct Solution;
impl Solution {
    pub fn count_subarrays(nums: Vec<i32>) -> i32 {
        let mut cnt = 0;
        for chunk in nums.windows(3) {
            if let Some(mid) = chunk[1].checked_div_euclid(2) {
                if (chunk[0] + chunk[2]) == mid && mid * 2 == chunk[1] {
                    cnt += 1;
                }
            }
        }
        cnt
    }
}
fn main() {
    println!("Hello, world!");
}
